Source/Modules/ModuleEditor.h

```diff
@@ -28,7 +28,7 @@ class ModuleEditor : public Module
 
 	void Resized();
 
-	void SetStateMachineWindowEditor(const std::weak_ptr<ResourceStateMachine>& resource);
+	void SetStateMachineWindowEditor(StateMachine*& resourceInstance, std::string& instanceName);
```

Comment:
Maybe a dumb question, but doesn't the signature make more sense like this:
```suggestion
	void SetStateMachineWindowEditor(StateMachine* resourceInstance, const std::string& instanceName);
```